File: RepoTransBench/source_projects/JavaScript/codemeasandwich_graphql-query-builder/public_tests/test.public.js
"use strict";
var expect = require('chai').expect;
var Query = require('../index');

function removeSpaces(textS) {
    return `${textS}`.replace(/\s+/g, '');
}

describe("graphql query builder (public cases)", function() {
    it('should accept a single find value (public)', function(){
        let expected = `product{price}`;
        let q = new Query("product").find("price");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should create a Query with function name & alias (public)', function(){
        let expected = `best : product{title}`;
        let q = new Query("product","best").find("title");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should create a Query with function name & input (public)', function(){
        let expected = `order(orderId:56789){status}`;
        let q = new Query("order",{orderId:56789}).find("status");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should create a Query with function name & input(s) (public)', function(){
        let expected = `order(orderId:56789, quantity:10){status}`;
        let q = new Query("order",{orderId:56789, quantity:10}).find("status");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should accept array as find argument (public)', function(){
        let expected = `shop{location, items}`;
        let q = new Query("shop").find(["location", "items"]);
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should handle nested find (public)', function(){
        let expected = `profile{id, data{email}}`;
        let q = new Query("profile").find(["id", {data: ["email"]}]);
        expect(removeSpaces(expected)).to.equal(removeSpaces(q));
    });

    it('should stringify toString correctly (public)', function(){
        let expected = `team{members}`;
        let q = new Query("team").find("members");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q.toString()));
    });

    it('should support toString with simple case (public)', function(){
        let expected = `item{name}`;
        let q = new Query("item").find("name");
        expect(removeSpaces(expected)).to.equal(removeSpaces(q.toString()));
    });
});

File: RepoTransBench/target_projects/JavaScript/Python/codemeasandwich_graphql-query-builder/public_tests/test_public.py
import pytest
from src.graphql_query_builder.index import Query

def remove_spaces(s):
    return str(s).replace(' ', '').replace('\t', '').replace('\n', '')

def test_should_accept_single_find_value_public():
    expected = "product{price}"
    q = Query("product").find("price")
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_create_query_with_function_name_and_alias_public():
    expected = "best : product{title}"
    q = Query("product", "best").find("title")
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_create_query_with_function_name_and_input_public():
    expected = "order(orderId:56789){status}"
    q = Query("order", {"orderId": 56789}).find("status")
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_create_query_with_function_name_and_inputs_public():
    expected = "order(orderId:56789, quantity:10){status}"
    q = Query("order", {"orderId": 56789, "quantity": 10}).find("status")
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_accept_array_as_find_argument_public():
    expected = "shop{location, items}"
    q = Query("shop").find(["location", "items"])
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_handle_nested_find_public():
    expected = "profile{id, data{email}}"
    q = Query("profile").find(["id", {"data": ["email"]}])
    assert remove_spaces(expected) == remove_spaces(q)

def test_should_stringify_to_string_correctly_public():
    expected = "team{members}"
    q = Query("team").find("members")
    assert remove_spaces(expected) == remove_spaces(q.to_string())

def test_should_support_to_string_with_simple_case_public():
    expected = "item{name}"
    q = Query("item").find("name")
    assert remove_spaces(expected) == remove_spaces(q.to_string())